node/pkg/fetcher/fetcher.go

```diff
@@ -1 +1,142 @@
 package fetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/bus"
+	"bisonai.com/orakl/node/pkg/db"
+	"bisonai.com/orakl/node/pkg/utils"
+)
+
+const FETCHER_FREQUENCY = 2 * time.Second
+
+func NewFetcher(bus *bus.MessageBus) *Fetcher {
+	return &Fetcher{
+		Adapters: make([]AdapterDetail, 0),
+		Bus:      bus,
+	}
+}
+
+func (f *Fetcher) Run(ctx context.Context) error {
+	err := f.initialize(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(FETCHER_FREQUENCY)
+	go func() {
+		for range ticker.C {
+			err := f.runAdapter(ctx)
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (f *Fetcher) runAdapter(ctx context.Context) error {
```

Comment:
I think `fetchAll` would describe the function better since `fetch` function collects data from a single adapter and in here we run it for all adapters.